cpu.go

```diff
@@ -273,17 +273,23 @@ func (ca *Cagent) CPUWatcher() *CPUWatcher {
 	// optimization to prevent CPU watcher to run in case CPU util metrics not are not needed
 	if len(ca.Config.CPUUtilTypes) > 0 && len(ca.Config.CPUUtilDataGather) > 0 || len(ca.Config.CPULoadDataGather) > 0 {
 		err := cw.Once()
+		if _, timeoutError := err.(TimeoutError); err == nil || timeoutError {
```

Comment:
I red this line 10 times and for me it's still not clear what the condition is here...
You check against `err == nil` and `err == timeoutError` in the same if branch. Please at least add a comment to explain what's going on here

Reply:
fixed

Reply:
I split it and added the comment

Reply:
ok, thanks for adding the comment.
Should we really run CPU Watcher continuously in case of timeout? After 3 or 5 timed out queries maybe we should stop and write a log statement?

Reply:
first, let's try to fix this issue then we could try to optimise
I don't know, maybe there is a chance that WMI DLL can be locked for 1 minute so all 6 attempts with a 10-second timeout will fail